proto/minder/v1/minder.proto

```diff
@@ -3411,3 +3412,144 @@ message UpstreamEntityRef {
+    string id = 5 [
+        (buf.validate.field).string = {uuid: true},
+        (google.api.field_behavior) = OUTPUT_ONLY
+    ];
+
+    // input is the input configuration for the data source.
+    // It allows us to configure a JSON schema for the input data.
+    google.protobuf.Struct input = 6;
+
+    oneof driver {
+        // rest is the REST data source driver.
+        RestDataSource rest = 7;
+    }
+}
+
+// RestDataSource is the REST data source driver.
+message RestDataSource {
+    // endpoint is the URL of the REST API.
+    string endpoint = 1 [
+        (buf.validate.field).string = {
+            pattern: "^https?://.*$",
+            max_len: 800,
+        }
+    ];
+
+    // method is the HTTP method to use for the request.
+    string method = 2 [
+        (buf.validate.field).string = {
+            in: ["GET", "POST", "PUT", "PATCH", "DELETE"]
+        },
+        (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+    ];
+
+    // headers is a map of headers to send with the request.
+    google.protobuf.Struct headers = 3;
+
+    oneof body {
+        // body is the body of the request.
+        google.protobuf.Struct bodyobj = 4;
+        // bodystr is the body of the request as a string.
+        string bodystr = 5 [
+            (buf.validate.field).string = {
+                max_len: 1000,
+            },
+            (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+        ];
+    }
+
+    // parse is the parse configuration for the response.
+    // This allows us to serialize the response into a structured format,
+    // or not.
+    // If left unset, the response will be treated as a string.
+    string parse = 6 [
+        (buf.validate.field).string = {
+        pattern: "^[a-z_]+$",
+            max_len: 10,
+            },
+        (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+    ];
+
+    message Fallback {
+        int32 http_status = 1 [
+            (buf.validate.field).int32 = {gte: 100, lte: 599},
+            (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+        ];
+
+        string body = 2 [
+            (buf.validate.field).string = {
+                max_len: 1000,
+            },
+            (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+        ];
+    }
+
+    // fallback is the fallback configuration for the response in case
+    // of an unexpected status code.
+    repeated Fallback fallback = 7;
+
+    // expected_status is the expected status code for the response.
+    // This may be repeated to allow for multiple expected status codes.
+    // If left unset, it will default to 200.
+    repeated int32 expected_status = 8 [
+        (buf.validate.field).repeated = {
+            items: {
+                int32: {
+                    gte: 100,
+                    lte: 599,
+                }
+            }
+        },
+        (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+    ];
+
+    DataSourceAuth auth = 9;
```

Comment:
Do we need this proto and field now, or can we add it later and the default empty value means "no auth"?

Reply:
We can add it later, no worries.